syntax = "proto2";

package client_messages;

////////////////////// ctx
message HandleClientRequest{
    required string request_type=1;
    required string key=2;
    optional string value=3;
	optional int64 requestid=4;	
	optional int64 expirerequestid=5;
    optional int64 clientid=6;
}

message HandleClientResponse{
    required int32 flag=1;
    optional string leader_name=2;  ///若操作请求的Node不是leader，则返回leader_name
    optional string value=3;  ///get操作返回的value
}

service ClientMessages {
    rpc HandleClient(HandleClientRequest) returns (HandleClientResponse) {}
}
